parser/prism/Translator.cc

```diff
@@ -4212,23 +4246,32 @@ NodeVec Translator::translateNumberedParametersNode(pm_numbered_parameters_node
     ENFORCE(1 <= paramCount, "A `pm_numbered_parameters_node_t` node should have at least one parameter");
     ENFORCE(paramCount <= 9, "Ruby only supports 9 numbered parameters (`_9` but no `_10`).");
 
+    auto numberedParamsUsageLocs = findNumberedParamsUsageLocs(location, statements, paramCount);
+    ENFORCE(paramCount <= numberedParamsUsageLocs.size());
+
     NodeVec params;
     params.reserve(paramCount);
 
     for (auto i = 1; i <= paramCount; i++) {
+        // Numbered parameters are implicit, so they don't have a real location in the body.
+        // However, we need somewhere for the error messages to point to, so we use the
+        // location of the first *usage* of this numbered parameter (or none if it was never used).
+        auto usageLoc = numberedParamsUsageLocs[i - 1];
         auto name = ctx.state.enterNameUTF8("_" + to_string(i));
 
-        // The location is arbitrary and not really used, since these aren't explicitly
```

Comment:
These locations are definitely used, in places like:

https://github.com/sorbet/sorbet/blob/a31b2c30e34c71e60c021e9886a1bd0f9eeb3faf/test/testdata/parser/error_recovery/forward_args.rb#L5-L6